src/persist-client/src/lib.rs

```diff
@@ -383,14 +369,17 @@ impl PersistClient {
                 Arc::clone(&self.metrics),
             )
         });
-        let shard_upper = machine.fetch_upper().await;
+        let writer_id = WriterId::new();
+        let (shard_upper, _) = machine.register_writer(&writer_id, SYSTEM_TIME()).await;
```

Comment:
this NowFn mock thing is an anti-pattern to me; even if we do end up needing to mock out the clock, it should be plumbed down by the unit tests instead of global static state. I'd just make `system_time()` pub and call it directly or, honestly probably better, just copy the impl into persist